junit5-support/src/main/kotlin/in/specmatic/test/reports/coverage/OpenApiCoverageReportInput.kt

```diff
@@ -0,0 +1,152 @@
+package `in`.specmatic.test.reports.coverage
+
+import `in`.specmatic.conversions.convertPathParameterStyle
+import `in`.specmatic.core.TestResult
+import `in`.specmatic.core.pattern.ContractException
+import `in`.specmatic.test.API
+import `in`.specmatic.test.TestResultRecord
+import kotlin.math.min
+import kotlin.math.roundToInt
+
+class OpenApiCoverageReportInput(
+    private val testResultRecords: MutableList<TestResultRecord> = mutableListOf(),
+    private val applicationAPIs: MutableList<API> = mutableListOf(),
+    private val excludedAPIs: MutableList<String> = mutableListOf()
+) {
+    fun addTestReportRecords(testResultRecord: TestResultRecord) {
+        testResultRecords.add(testResultRecord)
+    }
+
+    fun addAPIs(apis: List<API>) {
+        applicationAPIs.addAll(apis)
+    }
+
+    fun addExcludedAPIs(apis: List<String>){
+        excludedAPIs.addAll(apis)
+    }
+
+    fun generate(): OpenAPICoverageReport {
+        // If we get a failed test whose path and method are not found in the actuator, it means that the operation exists in the spec, but has not been implemented.
```

Comment:
Can we delete comment and extract this section into a method with a name that conveys the same? Example: identifyTestThatFailedBecauseOfEndpointsThatWereNotImplemented

Reply:
Fixed.